org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java

```diff
@@ -49,7 +54,15 @@ public void run() {
 		if (!parentProcessStillRunning()) {
 			JavaLanguageServerPlugin.logInfo("Parent process stopped running, forcing server exit");
 			task.cancel(true);
-			server.exit();
+			if (JavaLanguageServerPlugin.getInstance() != null && JavaLanguageServerPlugin.getInstance().getProtocol() instanceof org.eclipse.lsp4j.services.LanguageServer) {
+				((org.eclipse.lsp4j.services.LanguageServer) JavaLanguageServerPlugin.getInstance().getProtocol()).exit();
```

Comment:
Another question about this `if` statement. You want to ensure that `JavaLanguageServerPlugin.getInstance().getProtocol()` isn't `null` and is a `org.eclipse.lsp4j.services.LanguageServer`. However, could this ever be null ? `ParentProcessWatcher` is instantiated at 

https://github.com/eclipse/eclipse.jdt.ls/blob/9f5473609f9d194d37036a2818095cdc8b39b5ca/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java#L381

which is after `protocol` is defined at

https://github.com/eclipse/eclipse.jdt.ls/blob/9f5473609f9d194d37036a2818095cdc8b39b5ca/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java#L354-L358

Only case I can think of is maybe protection for the tests.

Reply:
> However, could this ever be null 

It can be null if JavaLanguageServerPlugin is stopped. See JavaLanguageServerPlugin.stop(BundleContext)